codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -378,16 +384,34 @@ protected Http2FrameWriter frameWriter() {
      * Writes a {@code RST_STREAM} frame to the remote endpoint and updates the connection state appropriately.
      */
     @Override
-    public ChannelFuture writeRstStream(ChannelHandlerContext ctx, int streamId, long errorCode,
-            ChannelPromise promise) {
-        Http2Stream stream = connection().stream(streamId);
+    public ChannelFuture writeRstStream(final ChannelHandlerContext ctx, int streamId, long errorCode,
+            final ChannelPromise promise) {
+        final Http2Stream stream = connection().stream(streamId);
+        if (stream == null) {
+            // Don't write a RST_STREAM frame if we are not aware of the stream.
+            return promise.setFailure(streamError(streamId, INTERNAL_ERROR,
+                    "Stream %d not found while writing RST_STREAM frame", streamId));
+        }
+        if (stream.isResetSent()) {
+            // Don't write a RST_STREAM frame if we already wrote a RST_STREAM frame
+            return promise.setFailure(streamError(streamId, INTERNAL_ERROR,
```

Comment:
@normanmaurer It says `RST_STREAM frames MUST NOT be sent for a stream in the "idle" state. `, but doesn't seem to impose any limitations on how often one can send a `RST_STREAM` frame or say something about other states: http://http2.github.io/http2-spec/#RST_STREAM

I mean sending multiple RST_STREAM frames should be idempotent. So that's why I think it makes sense to succeed the future here.